File: ExpoApp.Service/ViewModels/Expo/ExpoVM.cs
using ExpoApp.Core.Models;
using ExpoApp.Service.ViewModels.Industries;
using ExpoApp.Service.ViewModels.Location;
using ExpoApp.Service.ViewModels.Tickets;
using ExpoApp.Service.ViewModels.User;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ExpoApp.Service.ViewModels.Expo
{
    public class ExpoVM
    {
        public int Id { get; set; }
        public string Title { get; set; }
        public string Description { get; set; }
        public DateTime DateOfOpening { get; set; }
        public DateTime DateOfClosing { get; set; }
        public string WorkHoursOpening { get; set; }
        public string WorkHoursClosing { get; set; }
        public bool HasExpired { get { return DateOfClosing < DateTime.Now;  } }
        public OrganizerVM Organiser { get; set; }
        public LocationVM Location { get; set; }
        public IndustryVM Industry { get; set; }
        public List<TicketVM> Tickets { get; set; }
        public List<TicketTypeVM> TicketTypes { get; set; }
    }
}

File: ExpoApp.Web/src/app/models/ExpoVM.ts
import { IndustryVM } from './IndustryVM';
import { LocationVM } from './LocationVM';
import { TicketVM } from './TicketVM';

export interface ExpoVM {
  id: number;
  title: string;
  description: string;
  dateOfOpening: Date;
  dateOfClosing: Date;
  workHoursOpening: number;
  workHoursClosing: number;
  hasExpired: boolean;
  organizer: string;
  location: LocationVM;
  industry: IndustryVM;
  ticketTypes: TicketVM[];
  show:boolean;
}
